src/python/pants/backend/python/interpreter_cache.py

```diff
@@ -185,18 +185,18 @@ def setup(self, filters=()):
     filters = filters if any(filters) else self._python_setup.interpreter_constraints
     setup_paths = (self.pex_python_paths()
                    or self._python_setup.interpreter_search_paths
-                   or os.getenv('PATH').split(os.pathsep))
+                   or os.getenv(b'PATH').split(os.pathsep))
     logger.debug(
       'Initializing Python interpreter cache matching filters `{}` from paths `{}`'.format(
         ':'.join(filters), ':'.join(setup_paths)))
 
-    def unsatisfied_filters(interpreters):
-      return [f for f in filters if len(list(self._matching(interpreters, [f]))) == 0]
+    def unsatisfied_filters(ipreters):
```

Comment:
How about extracting a method instead of the tortured name de-shadowing?

Reply:
That seems like overkill, and would require passing `filters` in. However, I can remove the argument entirely and have the function look at `interpreters` directly from its closure.